### ISSUE TITLE
Kernel module loading errors are reported incorrectly

### ISSUE THREAD

**Issue by travisdowns** [obilaniu/libpfc#8]
Apparently kernel modules should return 0 (success) or a negative value (error code), but pfc.ko is instead returning 1, per the dmesg error output (noted in issue #7 ):

```
[50862.620450] do_init_module: 'pfc'->init suspiciously returned 1, it should follow 0/-E convention
               do_init_module: loading module anyway...
```

The side effect is that the kernel loads the module even though it has failed (i.e., 1 is treated as "success").

**Comment by obilaniu** [comment]
@travisdowns Wow, I have no idea how I closed this issue without clicking the button.

All I know is that my commit message had `Fix #8:` within it.

**Comment by travisdowns** [comment]
@obilaniu - it's a feature of GitHub. Certain phrases in the commit message auto-close issues:

https://help.github.com/articles/closing-issues-via-commit-messages/
